### PR TITLE
topics(grid): Adding external filterings for grid #1546

### LINKED ISSUE #1613: External Excel Style and Advanced filtering dialogs topics
Related to IgniteUI/igniteui-angular#5981

### PR DESCRIPTION
**Pull request by astaev**
Closes #1613

### CODE REVIEW

File: en/components/grids_templates/advanced_filtering.md [hunk 1]
```diff
@@ -110,6 +110,70 @@ In case you don't want to show the @@igComponent toolbar, you could use the [`op
 >[!NOTE]
 >You can enable both the [`quickFilter`]({environment:angularApiUrl}/enums/filtermode.html#quickfilter)/[`excelStyleFilter`]({environment:angularApiUrl}/enums/filtermode.html#excelstylefilter) and the advanced filtering user interfaces in the @@igComponent. Both filtering user interfaces will work independently of one another. The final filtered result in the @@igComponent is the intersection between the results of the two filters.
 
+
+### External Advanced filtering
+
+As you see the demo above the appearance of the Advanced filtering dialog is inside the Infragistics grid. When the setup in the dialog is ready, the apply or close actions would hide that dialog. There is a way to make that dialog stay always visible - it can  be used outside of the grid as a standalone component. At the demo below, the advanced filtering dialog is declared separately of the grid.
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/advanced_filtering.md]: - The advanced filtering dialog is not exactly inside the grid. You could say 'the Advanced filtering dialog is hosted in an overlay on top of the @@igComponent'.
- There is double space 'can  be'.
- Modify 'be used outside of the grid as a standalone component' to 'be used as a standalone component'.
- '**At** the demo below' should be '**In** the demo below'.
- 'separately of the grid' should be 'separately of the @@igComponent'.

File: en/components/grids_templates/advanced_filtering.md [hunk 2]
```diff
@@ -110,6 +110,70 @@ In case you don't want to show the @@igComponent toolbar, you could use the [`op
...
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-sample-1-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+}
+
+#### Usage
+
+It's super easy to configure the advanced filtering to work outside of the grid. All you need to do is to create the dialog and set its `grid` property, then apply some filters:
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/advanced_filtering.md]: - 'outside of the grid' should be 'outside of the @@igComponent'
- Add a link to the API docs of the `grid` property.
- Remove the ', then apply some filters' part.

File: en/components/grids_templates/excel_style_filtering.md [hunk 3]
```diff
@@ -450,6 +450,94 @@ In order to provide a custom loading template for the excel style filtering, we
 
 <div class="divider--half"></div>
 
+
+### External Excel Style filtering
+
+As you see at the demos above the default appearance of the Excel Style filtering dialog is inside the grid. So this dialog is only visible when configuring the fitlers. There is a way to make that dialog stay always visible - it can  be used outside of the grid as a standalone component. All you need to do is to add it in your html and set its `column` property.
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/excel_style_filtering.md]: - 'inside the grid' should be 'inside the @@igComponent'.
- Remove double space 'can  be'.
- The last sentence "All you need..." seems to be repeated in other places. You could replace it here with something like 'In the demo below, the Excel style filtering is declared separately of the @@igComponent'.

File: en/components/grids_templates/excel_style_filtering.md [hunk 4]
```diff
@@ -450,6 +450,94 @@ In order to provide a custom loading template for the excel style filtering, we
...
+   </igx-select-item>
+</igx-select>
+
+<igx-grid-excel-style-filtering [column]="hierarchicalgrid1.getColumnByName(gridColums.value)">
+</igx-grid-excel-style-filtering>
+```
+}
+
+
+The excel style filtering `column` is set via the igx-select selected value, which is actually contains the grid columns. That's all you need to make the excel style filtering dialog work outside of the grid, as a standalone component.
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/excel_style_filtering.md]: You could remove that paragraph.

File: en/components/grids_templates/excel_style_filtering.md [hunk 5]
```diff
@@ -450,6 +450,94 @@ In order to provide a custom loading template for the excel style filtering, we
...
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="grid-sample-1-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+}
+
+
+<div class="divider--half"></div>
+
+#### Usage
+
+At the sample above a column should be selected first, in order to instruct the ExcelStyle filter dialog which grid's column to filtrate. You can filter, hide or show, pin or unpin given column via that dialog. All you need to do, is to create the excel style dialog and bind its column property to a grid's column:
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/excel_style_filtering.md]: Replace that paragraph with something like 'In order to configure the Excel style filtering component, you should set its `column`(API link) property to one of the @@igComponent's columns. In the sample above, we have bound the `column`(API link) property to the `value` of an `IgxSelectComponent` that displays the @@igComponent's columns.'

File: en/components/grids_templates/advanced_filtering.md (hunk 2 above)
Comment (astaev) [reply en/components/grids_templates/advanced_filtering.md]: Now sure what's the url...

Comment (DiyanDimitrov) [reply en/components/grids_templates/advanced_filtering.md]: The url should be {environment:angularApiUrl}/classes/igxadvancedfilteringdialogcomponent.html#grid

File: en/components/grids_templates/excel_style_filtering.md (hunk 5 above)
Comment (astaev) [reply en/components/grids_templates/excel_style_filtering.md]: Not sure about the API link...

File: en/components/grids_templates/advanced_filtering.md (hunk 2 above)
Comment (astaev) [reply en/components/grids_templates/advanced_filtering.md]: Thanks

File: en/components/grids_templates/advanced_filtering.md [hunk 6]
```diff
@@ -110,6 +110,70 @@ In case you don't want to show the @@igComponent toolbar, you could use the [`op
 >[!NOTE]
 >You can enable both the [`quickFilter`]({environment:angularApiUrl}/enums/filtermode.html#quickfilter)/[`excelStyleFilter`]({environment:angularApiUrl}/enums/filtermode.html#excelstylefilter) and the advanced filtering user interfaces in the @@igComponent. Both filtering user interfaces will work independently of one another. The final filtered result in the @@igComponent is the intersection between the results of the two filters.
 
+
+### External Advanced filtering
+
+As you see the demo above the appearance of the Advanced filtering dialog is hosted in an overlay on top of the @@igComponent. When the setup in the dialog is ready, the apply or close actions would hide that dialog. There is a way to make that dialog stay always visible - be used as a standalone component. In the demo below, the advanced filtering dialog is declared separately of the @@igComponent.
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/advanced_filtering.md]: Remove the 'the appearance of ' part.

File: en/components/grids_templates/excel_style_filtering.md [hunk 7]
```diff
@@ -341,6 +341,91 @@ If you want to keep the sorting, moving, pinning and hiding features of the colu
...
+}
+
+@@if (igxName === 'IgxHierarchicalGrid') {
+<div class="sample-container loading" style="height:670px">
+    <iframe id="hierarchicalgrid-sample-3-iframe" src='{environment:demosBaseUrl}/hierarchical-grid/hierarchical-grid-external-excel-style-filtering' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" disabled class="stackblitz-btn" data-iframe-id="hierarchicalgrid-sample-3-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
```
Comment (DiyanDimitrov) [inline en/components/grids_templates/excel_style_filtering.md]: Duplicated iframe ids.

**Pull request merged**